common/djangoapps/student/views.py

```diff
@@ -1710,7 +1715,10 @@ def create_account_with_params(request, params):
     # Perform operations within a transaction that are critical to account creation
     with transaction.atomic():
         # first, create the account
-        (user, profile, registration) = _do_create_account(form, custom_form)
+        try:
+            (user, profile, registration) = _do_create_account(form, custom_form)
+        except PermissionDenied:
```

Comment:
In RegistrationView post api (openedx/core/djangoapps/user_api), create_account_with_params() is called to create User, so PermissionDenied has been raised in create_account_with_params() to catch it properly in user_api